public/src/components/admin/MarketState.tsx

```diff
@@ -0,0 +1,117 @@
+          // showNotif("Market has been successfully!");
+          this.setState(prevState => {
+            return {
+                option: false
+            }
+          });
+        }catch(e){
+          console.log("Error happened while Opening Market! ", e.statusCode, e.statusMessage, e);
+          if (e.isGrpcError) {
+            showErrorNotif("Oops! Unable to reach server. Please check your internet connection!");
+          } else {
+            showErrorNotif("Oops! Something went wrong! " + e.statusMessage);
+          }
+        }
+    }
+ 
+    handleCloseMarket = async (e:any) =>{
+        const option = this.state.option;
+        const sessionMd = this.props.sessionMd;
+        // close Market
+        const MarketRequest = new CloseMarketRequest();
+        try{
+          MarketRequest.setUpdatePrevDayClose(option);
+          const resp = await DalalActionService.closeMarket(MarketRequest, sessionMd);
+          // If any error occurs, it will be raised in DalalMessage_pb_Service
+          showNotif("Market has been successfully closed!");
+          this.setState(prevState => {
+            return {
+                option: false
+            }
+          });
+        }catch(e){
+          console.log("Error happened while Closing Market! ", e.statusCode, e.statusMessage, e);
+          if (e.isGrpcError) {
+            showErrorNotif("Oops! Unable to reach server. Please check your internet connection!");
+          } else {
+            showErrorNotif("Oops! Something went wrong! " + e.statusMessage);
+          }
+        }
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <table id="MarketState">
+                    <tbody className="ui bottom attached tab segment active inverted">
+                    <tr>
+                      <td>
+                        <label>{this.props.isMarketOpen ? "Do you want to update Previous Day Close ?" : "Do you want to update Day`s high and low ?"}</label>
+                      </td>
+                      <td>
+                        <label className="radiolabel">
+                         <input type="radio" id="true" onChange={(e) => {this.handleOptionChange()}} checked={this.state.option === true}/>Yes
+                         <span className="radiocheckmark"></span>
+                        </label>
+                        <label className="radiolabel">
+                         <input type="radio" id="false" onChange={(e) => {this.handleOptionChange()}} checked={this.state.option == false}/>No
+                         <span className="radiocheckmark"></span>
+                        </label>
+                      </td>
+                    </tr>
+                    <tr>
+                      <td>
+                       <input type="button" disabled={this.props.isMarketOpen ? true : false} className={"ui inverted green button"} onClick={(e) => {this.handleOpenMarket(e)}} value="Open Market"/>
```

Comment:
Since boolean values of `this.props.isMarketOpen` corresponds to true and false itself, the conditional statement is not needed here.